spring-integration-core/src/main/java/org/springframework/integration/channel/ReactiveStreamsSubscribableChannel.java

```diff
@@ -20,14 +20,39 @@
 
 import org.springframework.messaging.Message;
 
+import reactor.core.publisher.Mono;
+
 /**
+ * Reactive Streams specific message channel for composing upstream {@link Publisher}s
+ * into an on-demand processing in this channel implementation.
+ *
  * @author Artem Bilan
  * @author Gary Russell
  *
  * @since 5.0
  */
 public interface ReactiveStreamsSubscribableChannel {
 
+	/**
+	 * Subscribe to the provided {@link Publisher}.
+	 * @param publisher the {@link Publisher} to subscribe to.
+	 * @deprecated in favor of {@link #subscribeToUpstream}.
+	 * Framework doesn't call this method internally any more.
+	 */
+	@Deprecated
 	void subscribeTo(Publisher<? extends Message<?>> publisher);
 
+	/**
+	 * Compose with the provided upstream {@link Publisher} for processing its events in this channel.
+	 * It is recommended to implement this method for possible reactive streams composition.
+	 * @param upstreamPublisher the upstream {@link Publisher} to subscribe to.
+	 * @return the {@link Mono} for reactive streams composition.
+	 * @since 5.2.2
+	 */
+	@SuppressWarnings("deprecagtion")
+	default Mono<Void> subscribeToUpstream(Publisher<? extends Message<?>> upstreamPublisher) {
+		subscribeTo(upstreamPublisher);
```

Comment:
I would highly recommend to subscribe only when the returned Mono is subscribed, otherwise you're creating a hot publisher